### DISCUSSION TITLE
Rendering to a portal

### DISCUSSION THREAD

**Discussion by rimzzlabs** [elrumordelaluz/reactour#586] [Q&A] [answered]
Is there any solution to render the tour component into a portal?
I've tried this way, but once the tour ends, and I tried to trigger it again, the tour did not mounted

```tsx
 <TourProvider
        scrollSmooth
        showBadge={false}
        showDots={false}
        showCloseButton={false}
        steps={tourSteps}
        ContentComponent={ContentComponent}
        Wrapper={(props: PropsWithChildren) =>
          createPortal(props.children, document.body)
        }
      />
```

**Comment by elrumordelaluz** [comment, marked as answer]
Hi @rizkimcitra, thanks for open the _Issue_.
The [demo](https://reactour.vercel.app/) is using a [Wrapper](https://github.com/elrumordelaluz/reactour/blob/main/apps/web/pages/index.tsx#L37), which is using a [Portal](https://github.com/elrumordelaluz/reactour/blob/main/apps/web/components/Portal.tsx).

Please let me know if this approach works on your end.

**Reply by rimzzlabs** [reply]
thanks for your reply, works based on your example now
